### PR TITLE
Multi Shop Product <-> Cart interaction bug fixes

### PR DESCRIPTION
**Pull request by sleroux**
This PR fixes a variety of issues with the Cart View:

* CartView now properly animates in and out in the tester scene's multi product shop.
* Fixes background colors for most of the panels including the Cart View
* Fixes bug with toggling of empty state on the Cart
* Fixes Unity crashing issue when trying to make reference to Shop property in CartView
* Small refactor in how we invalidate the Cart View and handle actions for increasing/decreasing

### CODE REVIEW

**Comment by richardmonette** [comment]
- [ ] 🎩 issue: when clicking the +/- buttons the product photo is flickering

**Comment by richardmonette** [comment]
- [ ] 🎩 issue: probably not in scope for this PR, but there is no indication that a product has been added to cart when clicking the Add to cart button

File: Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs [hunk 1]
```diff
@@ -38,15 +41,20 @@ public class CartItemsChangeEvent : UnityEvent<List<CartItem>> {}
...
+        private Dictionary<string, ProductAndVariantPair> _idsToProductPairs = new Dictionary<string, ProductAndVariantPair>();
 
         public CartController(Cart cart) {
             SetCart(cart);
         }
 
         public void SetCart(Cart cart) {
             Cart = cart;
-            _idsToVariants = new Dictionary<string, ProductVariant>();
+            _idsToProductPairs = new Dictionary<string, ProductAndVariantPair>();
```
Comment (richardmonette) [inline Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs]: Would `Clear` https://msdn.microsoft.com/en-us/library/b5txwy7s(v=vs.110).aspx better communicate the intent here?

File: Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs [hunk 2]
```diff
@@ -38,15 +41,20 @@ public class CartItemsChangeEvent : UnityEvent<List<CartItem>> {}
         public UnityEvent OnPurhcaseCancelled = new UnityEvent();
         public PurchaseFailedEvent OnPurhchaseFailed = new PurchaseFailedEvent();
 
-        private Dictionary<string, ProductVariant> _idsToVariants = new Dictionary<string, ProductVariant>();
+        private struct ProductAndVariantPair {
```
Comment (richardmonette) [inline Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs]: Once a pair is made, should it be mutable? If no, should we implement this with a constructor and getters? I think that would enforce the ability to make a pair, but not change it afterwards.

File: Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs [hunk 3]
```diff
@@ -161,14 +173,17 @@ public class CartItemsChangeEvent : UnityEvent<List<CartItem>> {}
             return Cart.LineItems.All().Sum((x) => (int) x.Quantity);
         }
 
-        private void CacheVariantByID(ProductVariant variant) {
-            if (!_idsToVariants.ContainsKey(variant.id())) {
-                _idsToVariants.Add(variant.id(), variant);
+        private void CacheProductVariantPair(ProductVariant variant, Product product) {
+            if (!_idsToProductPairs.ContainsKey(variant.id())) {
+                var pair = new ProductAndVariantPair();
```
Comment (richardmonette) [inline Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs]: Thinking maybe this should be a parameterized constructor

**Comment by sleroux** [comment]
@richardmonette I've address the code comments and resolved the image flickering bug. The game objects for the list items would all be destroyed every time any of the cart items changed which caused the flicker. I've added some caching of the game objects so they only get added/removed when the number of cart items change.

**Comment by sleroux** [comment]
> probably not in scope for this PR, but there is no indication that a product has been added to cart when clicking the Add to cart button

This will be addressed in another PR by updating a counting badge icon on the cart button.

**Pull request merged**